src/main/java/com/gtnewhorizon/gtnhlib/CommonProxy.java

```diff
@@ -49,4 +54,18 @@ public void addErrorToChat(IChatComponent componentText) {}
     public void addMessageToChat(IChatComponent componentText) {}
 
     public void printMessageAboveHotbar(String message, int displayDuration, boolean drawShadow, boolean shouldFade) {}
+
+    /**
+     * Sends packet from server to client that will display message above hotbar.
+     * @see ClientProxy#printMessageAboveHotbar
+     */
+    public void sendMessageAboveHotbar(
+            EntityPlayerMP player,
+            IChatComponent chatComponent,
+            int displayDuration,
+            boolean drawShadow,
+            boolean shouldFade) {
+        NetworkHandler.instance.sendTo(
```

Comment:
As a precaution, please rule out fake player here.

Reply:
Also remember TC doesn’t extend FakePlayer.

Reply:
sure

Reply:
@miozune still doesn’t cover TC fake players.

Reply:
Now I think, is this check really needed? For example GG does send message without FakePlayer check:
https://github.com/GTNewHorizons/GoodGenerator/blob/bd60bd3e96368d1fe726cf501ce1166b397d896a/src/main/java/goodgenerator/blocks/tileEntity/GTMetaTileEntity/NeutronSensor.java#L102